src/openms/source/FORMAT/MzTab.cpp

```diff
@@ -1634,769 +1677,858 @@ namespace OpenMS
-      // and also: Why do we go through all runs here, while above for the search engine settings we only take the first.
-      // nonsense.
-      Size psm_search_engine_index(1);
-      for (auto const & se : search_engine_to_runs) // loop over (unique) search engine names
-      {
-        MzTabParameter psm_score_type;
-        const tuple<String, String, String>& name_version_score = se.first;
-        psm_score_type.fromCellString("[,," + get<0>(name_version_score) + " " + get<2>(name_version_score) + "," + get<1>(name_version_score) + "]");
-        meta_data.psm_search_engine_score[psm_search_engine_index] = psm_score_type;
-        meta_data.peptide_search_engine_score[psm_search_engine_index] = psm_score_type; // same score type for peptides
-        psm_search_engine_index++;
-      }
+  boost::optional<MzTabPSMSectionRow> MzTab::PSMSectionRowFromPeptideID_(
+     const PeptideIdentification& pid,
+     const vector<const ProteinIdentification*>& prot_ids,
+     map<String, size_t>& idrun_2_run_index,
+     map<pair<size_t,size_t>,size_t>& map_run_fileidx_2_msfileidx,
+     map<Size, vector<pair<String, String>>>& run_to_search_engines,
+     const int psm_id,
+     const MzTabString& db,
+     const MzTabString& db_version,
+     const bool export_empty_pep_ids) 
+  {
+    // skip empty peptide identification objects, if they are not wanted
+    if (pid.getHits().empty() && !export_empty_pep_ids)
+    {
+      return boost::none;
+    }
+
+    /////// Information that doesn't require a peptide hit ///////
+    MzTabPSMSectionRow row;
+    row.PSM_ID = MzTabInteger(psm_id);
+    row.database = db;
+    row.database_version = db_version;
+    
+    vector<MzTabDouble> rts_vector;
+    rts_vector.emplace_back(MzTabDouble(pid.getRT()));
+
+    MzTabDoubleList rts;
+    rts.set(rts_vector);
+    row.retention_time = rts;
+
+    row.exp_mass_to_charge = MzTabDouble(pid.getMZ());
 
-      // map (indist.)protein groups to their protein hits (by index).
-      map<Size, set<Size>> ind2prot; // indistinguishable protein groups
-      map<Size, set<Size>> pg2prot; // general protein groups
+    // meta data on peptide identifications
+    vector<String> pid_keys;
+    pid.getKeys(pid_keys);
+    replaceWhiteSpaces_(pid_keys.begin(), pid_keys.end());
 
-      const std::vector<ProteinHit> proteins = prot_ids.front().getHits();
+    set<String> pid_key_set(pid_keys.begin(), pid_keys.end());
+    addMetaInfoToOptionalColumns(pid_key_set, row.opt_, String("global"), pid);
 
-      // map indistinguishable groups to the contained proteins
-      const std::vector<ProteinIdentification::ProteinGroup>& indist_groups = prot_ids.front().getIndistinguishableProteins();
-      Size ind_idx{0};
-      for (const ProteinIdentification::ProteinGroup & p : indist_groups)
+    // link to spectrum in MS run
+    String spectrum_nativeID = pid.getMetaValue("spectrum_reference").toString();
+    size_t run_index = idrun_2_run_index[pid.getIdentifier()];
```

Comment:
I know it was probably already in the old code but maybe (especially since it is in a separate function now) we should use .at() in this part to not get any bad surprises.
Probably not very speed-critical here.